Trading/Exchange/binanceusdm/binanceusdm_exchange.py

```diff
@@ -0,0 +1,365 @@
+#  Drakkar-Software OctoBot-Tentacles
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+
+import decimal
+import typing
+import ccxt as ccxt
+
+import octobot_trading.errors as trading_errors
+import octobot_trading.exchanges as exchanges
+import octobot_trading.enums as trading_enums
+import octobot_trading.exchanges.connectors.ccxt.enums as ccxt_enums
+import octobot_trading.constants as constants
+import octobot_commons.constants as commons_constants
+
+
+class BinanceUsdM(exchanges.RestExchange):
+    DESCRIPTION = ""
+    MARK_PRICE_IN_TICKER = True
+    FUNDING_IN_TICKER = True
+
+    @classmethod
+    def get_supported_exchange_types(cls) -> list:
+        return [trading_enums.ExchangeTypes.FUTURE]
+
+    def get_adapter_class(self):
+        return BinanceUsdMAdapter
+
+    @classmethod
+    def get_name(cls):
+        return "binanceusdm"
+
+    async def get_price_ticker(
+        self, symbol: str, **kwargs: dict
+    ) -> typing.Optional[dict]:
+        try:
+            with self.connector.error_describer():
+                return self.connector.adapter.adapt_ticker(
+                    {
+                        **await self.connector.client.fetch_ticker(
+                            symbol, params=kwargs
+                        ),
+                        # fetch_funding_rate to get mark price with funding
+                        **await self.connector.client.fetch_funding_rate(
+                            symbol, params={}
+                        ),
+                    }
+                )
+        except ccxt.NotSupported:
+            raise trading_errors.NotSupported
+        except ccxt.BaseError as e:
+            raise trading_errors.FailedRequest(f"Failed to get_price_ticker {e}")
+
+    async def set_symbol_partial_take_profit_stop_loss(
+        self,
+        symbol: str,
+        inverse: bool,
+        tp_sl_mode: trading_enums.TakeProfitStopLossMode,
+    ):
+        # no partial tp / sl - use limit and stop market instead
```

Comment:
Should we raise here to avoid giving the impression that it worked?

Reply:
oh yeah now we raise NotImplementedError